### PR TITLE
Fix donations that withdraw from trace with campaign recipient

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to #49, Giveth/giveth-dapp#2289

The logic works based on parent donations, so if you have no donations in your db and you run the simulation script, you should run it one or two times to parent donation be all set correctly, then after that the donations `isReturn` and `status` will work ok

### CODE REVIEW

File: src/repositories/donationRepository.ts [hunk 1]
```diff
@@ -1,20 +1,65 @@
...
+  }
+  const parentDonation = await findParentDonation(donation.parentDonations);
+  if (!parentDonation ||
+    parentDonation.status !== DonationStatus.PAID ||
+    parentDonation.ownerType !== AdminTypes.TRACE) {
+    return false;
+  }
+  const grandParentDonation = await findParentDonation(donation.parentDonations);
+  return Boolean(grandParentDonation &&
+    grandParentDonation.ownerType !== AdminTypes.CAMPAIGN &&
```
Comment (aminlatifi) [inline src/repositories/donationRepository.ts]: These two lines conditions should be === not !==

File: src/index.ts [hunk 2]
```diff
@@ -336,8 +344,27 @@ const handleToDonations = async ({
...
+    // If we didnt find any donation in above statement we should check to see if we find
+    // donations with isReturn===true, so in this case it's ok to donation have different isReturn
+    // if it's a donation with the trace donation parent, and it's ownerType is campaign
+    const toIndex= toNotFilledDonationList.findIndex(
+      item => item.txHash === transactionHash &&
+        item.amountRemaining === '0' &&
+        item.isReturn === true,
+    );
+    toDonation = toIndex !== -1 ? toNotFilledDonationList.splice(toIndex, 1)[0] : undefined;
+    const isDonationBackToCampaign = toDonation && await isDonationBackToCampaignFromTrace(toDonation)
```
Comment (aminlatifi) [inline src/index.ts:358]: We computed this logic (lines 358:364) once here and once in lines 567:571

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/repositories/donationRepository.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/repositories/donationRepository.ts]: @aminlatifi  resolve in #52

File: src/index.ts (hunk 2 above)
Comment (mohammadranjbarz) [reply src/index.ts:358]: @aminlatifi  There is two different logic, in L358 we check when no toDonaiton found, but in L567, we check when toDonation found, so it's not duplicated

Comment (aminlatifi) [reply src/index.ts:358]: @mohammadranjbarz you are right, thanks.

**Pull request merged**